### PR TITLE
Improvements and fixes for the docker development experience

### PR DESCRIPTION
**Pull request by cbwebdevelopment**
Goes hand-in-hand with the corresponding `clintb/development-build-fixes` branch in these repos:
https://github.com/tidepool-org/blip
https://github.com/tidepool-org/tools

See this trello card for any additional details:
https://trello.com/c/PnIamSBO

### CODE REVIEW

**Comment by cbwebdevelopment** [comment]
[Improvements and fixes for the docker development experience](https://trello.com/c/PnIamSBO/228-improvements-and-fixes-for-the-docker-development-experience)

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+# Common ENV
+ENV PORT=3000
+
+WORKDIR /app
+
+COPY package.json /app/package.json
+RUN apk add --no-cache --virtual .build-deps curl git \
+ && echo "Fixing PhantomJS to run on alpine" \
+ && curl -Ls "https://github.com/dustinblackman/phantomized/releases/download/2.1.1/dockerized-phantomjs.tar.gz" | tar xz -C / \
```
Comment (pazaan) [inline Dockerfile]: Should use the dependencies in `tools` repo.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ && echo "Re-adding node group and user, removed from PhantomJS fix" \
+ && addgroup -g 1000 node \
+ && adduser -u 1000 -G node -s /bin/sh -D node \
+ && yarn install \
```
Comment (pazaan) [inline Dockerfile]: This build fails. 

```
npm info lifecycle @tidepool/viz@0.8.3~postclean: @tidepool/viz@0.8.3
npm info ok
module.js:471
    throw err;
    ^

Error: Cannot find module '/app/package.config.js'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at module.exports (/app/node_modules/webpack/bin/convert-argv.js:80:13)
    at Object.<anonymous> (/app/node_modules/webpack/bin/webpack.js:39:40)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
```

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,24 @@
+FROM node:6.10.3-alpine
+
+WORKDIR /app
+
+COPY package.json /app/package.json
```
Comment (pazaan) [inline Dockerfile]: You could consolidate the 3 separate COPYs into:
`COPY package.json package.config.js webpack.config.js /app/`
Optional, not a blocker.

Comment (cbwebdevelopment) [reply Dockerfile]: Yes.  Good call.

**Pull request merged**